cmd/krew/cmd/update.go

```diff
@@ -59,59 +58,77 @@ func showFormattedPluginsInfo(out io.Writer, header string, plugins []string) {
 	fmt.Fprintf(out, "%s", b.String())
 }
 
-func showUpdatedPlugins(out io.Writer, preUpdate, posUpdate []index.Plugin, installedPlugins map[string]string) {
-	var newPlugins []index.Plugin
-	var updatedPlugins []index.Plugin
+func showUpdatedPlugins(out io.Writer, preUpdate, postUpdate []pluginEntry, installedPlugins map[string]string) {
+	var newPlugins []pluginEntry
+	var updatedPlugins []pluginEntry
 
-	oldIndex := make(map[string]index.Plugin)
+	oldIndexMap := make(map[string]pluginEntry)
 	for _, p := range preUpdate {
-		oldIndex[p.Name] = p
+		oldIndexMap[canonicalName(p.p, p.indexName)] = p
 	}
 
-	for _, p := range posUpdate {
-		old, ok := oldIndex[p.Name]
+	for _, p := range postUpdate {
+		cName := canonicalName(p.p, p.indexName)
+		old, ok := oldIndexMap[cName]
 		if !ok {
 			newPlugins = append(newPlugins, p)
 			continue
 		}
 
-		if _, ok := installedPlugins[p.Name]; !ok {
+		if _, ok := installedPlugins[cName]; !ok {
 			continue
 		}
 
-		if old.Spec.Version != p.Spec.Version {
+		if old.p.Spec.Version != p.p.Spec.Version {
 			updatedPlugins = append(updatedPlugins, p)
 		}
 	}
 
 	if len(newPlugins) > 0 {
 		var s []string
 		for _, p := range newPlugins {
-			s = append(s, p.Name)
+			s = append(s, displayName(p.p, p.indexName))
 		}
-
 		showFormattedPluginsInfo(out, "New plugins available", s)
 	}
 
 	if len(updatedPlugins) > 0 {
 		var s []string
 		for _, p := range updatedPlugins {
-			old := oldIndex[p.Name]
-			s = append(s, fmt.Sprintf("%s %s -> %s", p.Name, old.Spec.Version, p.Spec.Version))
+			old := oldIndexMap[canonicalName(p.p, p.indexName)]
```

Comment:
Minor nit, but I think the keys for `oldIndexMap` could use `displayName`. Then the line below could be brought up and this could just be `oldIndexMap[name]`.

Reply:
It's probably safer we just use canonicalName in internal parts of the code. displayName is only for displaying, and who knows may start colliding someday.